### PR TITLE
std.math: №2, fix abs, fabs

### PR DESCRIPTION
**Pull request by 9il**
intrinsics for `fabs` should be added to DMD.

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -3182,6 +3182,12 @@ real cbrt(real x) @trusted nothrow @nogc
  *      )
  */
 real fabs(real x) @safe pure nothrow @nogc;      /* intrinsic */
+//FIXME
```
Comment (kuettler) [inline std/math.d]: Sorry to be picky, but afaik there should be no `FIXME` in a pull request.

Comment (9il) [inline std/math.d]: No. I have already pushed scores of `FIXME`s into phobos. This is normal practise if DMD should be changed first.

**Comment by kuettler** [comment]
There is `fabsl`, `fabs` and `fabsf` in `core.stdc.math` whenever available. Maybe I just do not understand what you are trying to do.

**Comment by 9il** [comment]
This PRs have few goals:
1. fix dmd tests where code like `fabs(-2)` is used
2. fix unittest in phobos (same code style)
3. allow `fabs` and other intrinsics to be replaced in other compilers (like LDC)
4. allow to write other functions based on correct API: `float log(float)`, `double exp(double)`, `double gamma(double)`.

This isn't change for `fabs` itself, but for other functions and compilers.

**Comment by 9il** [comment]
PS: `core.stdc.math` is not `pure`.

**Comment by burner** [comment]
Could you add a unittest that calls all three version, simular to what is done in std.string

```D
foreach (S; TypeTuple!(float, double, real))
```

**Comment by 9il** [comment]
Unittest was added.

**Comment by burner** [comment]
Thank, LGTM

**Comment by burner** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
This pull request introduced a regression:
https://issues.dlang.org/show_bug.cgi?id=14842

**Comment by 9il** [comment]
... and this is good because implementation bugs like in `approxEqual` can be fixed faster both in Phobos and user code.

File: std/math.d (hunk 1 above)
Comment (WalterBright) [inline std/math.d]: Fix what?

Comment (9il) [inline std/math.d]: Fix code like
```D
double fabs(double x) @safe pure nothrow @nogc { return fabs(cast(real)x); }
```
to declaration
```D
double fabs(double x) @safe pure nothrow @nogc;
```
when DMD implement this intrinsic.

**Pull request merged**